src/DatabasesFileSystem.ts

```diff
@@ -0,0 +1,125 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { FileStat, FileType, MessageItem, Uri, workspace } from "vscode";
+import { AzExtItemQuery, AzExtItemUriParts, AzExtTreeFileSystem, AzExtTreeItem, DialogResponses, IActionContext, UserCancelledError } from 'vscode-azureextensionui';
+import { FileChangeType } from "vscode-languageclient";
+import { ext } from "./extensionVariables";
+import { MongoCollectionTreeItem } from "./mongo/tree/MongoCollectionTreeItem";
+import { localize } from "./utils/localize";
+import { getWorkspaceSetting, updateGlobalSetting } from "./utils/settingUtils";
+import { getNodeEditorLabel } from "./utils/vscodeUtils";
+
+export interface IEditableTreeItem extends AzExtTreeItem {
+    filePath: string;
+    cTime: number;
+    mTime: number;
+    getFileContent(context: IActionContext): Promise<string>;
+    writeFileContent(context: IActionContext, data: string): Promise<void>;
+}
+
+export class DatabasesFileSystem extends AzExtTreeFileSystem<IEditableTreeItem> {
+    public static scheme: string = 'azureDatabases';
+    public scheme: string = DatabasesFileSystem.scheme;
+    private _showSaveConfirmation: boolean = true;
+
+    public async statImpl(context: IActionContext, node: IEditableTreeItem): Promise<FileStat> {
+        const size: number = Buffer.byteLength(await node.getFileContent(context));
+        return { type: FileType.File, ctime: node.cTime, mtime: node.mTime, size };
+    }
+
+    public async readFileImpl(context: IActionContext, node: IEditableTreeItem): Promise<Uint8Array> {
+        return Buffer.from(await node.getFileContent(context));
+    }
+
+    public async writeFileImpl(context: IActionContext, node: IEditableTreeItem, content: Uint8Array, _originalUri: Uri): Promise<void> {
+        const showSavePromptKey: string = 'showSavePrompt';
+        const prefix: string = 'cosmosDB';
```

Comment:
Yeah I agree. I want the scheme for the file system to be generic so it can work for other resource types in the future, but I don't see enough reason (yet) to break backwards compatibility for people that have this setting. Added a note in the code to that effect